### PR TITLE
Code formatting and Django 4 ready

### PR DESCRIPTION
**Pull request by joshbrooks**
- Update code to be compatible with `black` and `flake8`
 - Update code to be compatible with Django 3.2 and address deprecationWarnings for Django 4
 - Adds github actions for black, flake8
 - Adds `pre-commit` for black, flake8

### CODE REVIEW

File: simple_locations/urls.py [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
-    url(r'^simple_locations/edit/(?P<area_id>[0-9]+)/$', edit_location),
-    url(r'^simple_locations/add/((?P<parent_id>[0-9]+)/){0,1}$', add_location),
-    url(r'^simple_locations/delete/(?P<area_id>[0-9]+)/$', delete_location),
-    url(r'^simple_locations/render_tree/$', render_location),
-    url(r'^simple_locations/$', simple_locations),
-    url(r'^areasearch/$', area_search),
+    path("simple_locations/edit/<int:area_id>/", edit_location),
+    path("simple_locations/add/", add_location),
+    path("simple_locations/add/<int:parent_id>/", add_location),
+    path("simple_locations/delete//<int:area_id>/", delete_location),
```
Comment (joshbrooks) [inline simple_locations/urls.py]:
```suggestion
    path("simple_locations/delete/<int:area_id>/", delete_location),
```

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+[tool.isort]
+profile = 'black'
+skip = ['env', 'migrations']
+
+[tool.black]
+line-length = 180
```
Comment (RobertoMaurizzi) [inline pyproject.toml]: This is the third different line lenght...😅

File: simple_locations/views.py [hunk 3]
```diff
@@ -1,31 +1,29 @@
-#!/usr/bin/python
-# -*- coding: utf-8 -*-
 from django.conf import settings
-from django.core import serializers
 from django.db import IntegrityError
 from django.http import HttpResponse, HttpResponseRedirect
-from django.shortcuts import render, get_object_or_404
+from django.shortcuts import get_object_or_404, render_to_response
```
Comment (RobertoMaurizzi) [inline simple_locations/views.py:4]: Why render_to_response instead of render? The former was sort-of replaced by the latter in Django 1.3...

**Review by RobertoMaurizzi — APPROVED** [review]
I'd like to try to import this in Liga Inan to see how much I'd need to refactor to have it working but there's a lot of better work that should come before that (like replacing health_service with a proper MPTT instance...)
Other than the very minor couple of things I commented on, I think it's good.

**Pull request closed without merging**